src/pages/dataElements/fields/NameField.tsx

```diff
@@ -0,0 +1,67 @@
+import i18n from '@dhis2/d2-i18n'
+import { InputFieldFF } from '@dhis2/ui'
+import React, { useMemo } from 'react'
+import { Field as FieldRFF, useField } from 'react-final-form'
+import { useParams } from 'react-router-dom'
+import {
+    composeAsyncValidators,
+    required,
+    useCheckMaxLengthFromSchema,
+    useIsFieldValueUnique,
+} from '../../../lib'
+import type { Validator } from '../../../lib'
+import type { SchemaName } from '../../../types'
+import type { FormValues } from '../form'
+
+type NameValidator = Validator<string, FormValues>
+
+function useValidator() {
+    const params = useParams()
+    const dataElementId = params.id as string
+    const checkIsValueTaken = useIsFieldValueUnique({
+        model: 'dataElements',
+        field: 'name',
+        id: dataElementId,
+    })
+
+    const checkMaxLength = useCheckMaxLengthFromSchema(
+        'dataElement' as SchemaName,
```

Comment:
Change import to `import { SchemaName} `, and use `SchemaName.dataElement` instead?